### DISCUSSION TITLE
Admin view and/or control panel

### DISCUSSION THREAD

**Discussion by FalkZerd** [pallets-eco/flask-security#574] [Q&A] [answered]
Im wondering if there is any kind of admin view to administrate users, or to register new users without the register form (for closed apps)? or is planned in a future?

also, trying to activate user from a `routes.py` and im totally lost

im trying 
```
@bp.route('/desactivar/<int:id>', methods=['GET', 'POST'])
@auth_required() 
@roles_accepted("Superuser")
def desactivar(id): #id from user
    try:
        user = User.query.get_or_404(id)
        security.user_datastore.deactivate_user(user)
        flash(f'user {user.name} is activated', 'success')

  
except Exception as e:
        print(e)
        flash('ups something is wrongl', 'danger')
    return redirect(request.referrer)
```

and ` __init___.py` 

```
db = SQLAlchemy()
#creating the application factory
fsqla.FsModels.set_db_info(db)
security = Security()

def create_app(config=DevelopmentConfig):
    app = Flask(__name__)
    app.config.from_object(config)

    from .admin.models import User, Role
    from .admin.forms import ExtendedLoginForm
   #initialization of extension instances
    db.init_app(app)
    user_datastore = SQLAlchemyUserDatastore(db, User, Role)
    security.init_app(app, user_datastore, login_form=ExtendedLoginForm)
```

and this error is happening all the time
  
`'Security' object has no attribute 'user_datastore'`

**Comment by jwag956** [comment, marked as answer]
A couple things going on. First - Flask-Security has stayed away from providing an admin UI interface - in general folks use Flask-Admin for that. Flask-Security does ship with a Flask CLI module - and you can create/delete users etc with that.

In terms of adding your own admin interface - that is pretty easy as well.

1. Use security.datastore.xxx
2. In general always use security.datastore.find_user and other helpers - as documented here: https://flask-security-too.readthedocs.io/en/stable/api.html#datastores. They properly update the database models.
3. Since you are using SQLAlchemy - make sure that for any mutating operation you call datastore.commit() at the end of the request.

**Comment by FalkZerd** [comment]
you dont have how much you helped me, I have been diging in the full documentation and examples trying to fix that, and now I feel like an idiot for its simplicity

thank you very much
